torch/utils/mkldnn.py

```diff
@@ -1,43 +1,149 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
-import functools
 
 import torch
 
 
-def to_mkldnn(module):
-    def t_fn(t):
-        if t.is_floating_point():
-            return t.to_mkldnn()
+class MkldnnLinear(torch.jit.ScriptModule):
+    def __init__(self, dense_module):
+        super(MkldnnLinear, self).__init__()
+        self.register_buffer('weight', dense_module.weight.to_mkldnn())
+        if dense_module.bias is not None:
+            self.register_buffer('bias', dense_module.bias.to_mkldnn())
+        else:
+            # TODO: Remove this once ScriptModule supports registering None buffer
```

Comment:
is there an issue for it?

Reply:
ScriptModule has overridden `register_buffer` to the C `_register_buffer` (same for *_parameter) which doesn't accept None as the argument. IIUC other script modules [store None as attribute](https://github.com/pytorch/pytorch/blob/796e359/torch/jit/__init__.py#L1534-L1536) to work around it. However, since this module has `__getstate__` and `__setstate__`, `export` will [skip serialization of any attributes](https://github.com/pytorch/pytorch/blob/796e3596014b0f684645aa58cb0fbd806a65b070/torch/csrc/jit/export.cpp#L853-L864).

Reply:
it's fine that we're skipping serialization of attributes. So what prevents us making it an attribute and handling it explicitly in `__get_state__/__set_state__`? or is there a specific reason it has to be a buffer?

and my original question was whether there's a GH issue to track supporting nullable buffers

Reply:
> my original question was whether there's a GH issue to track supporting nullable buffers

There is one, but in this lockdown repo.

Reply:
>  So what prevents us making it an attribute and handling it explicitly in __get_state__/__set_state__? or is there a specific reason it has to be a buffer?

So I don't think it has to be a buffer, the problem I had run into is the **parser** complains about `bias` is not an attribute when it parses the forward function that uses `self.bias`, no matter what the __getstate__ and __setstate__ do (as long as they exist).